notes-service/src/main/java/org/exoplatform/wiki/service/impl/NoteServiceImpl.java

```diff
@@ -428,18 +446,21 @@ public Page getNoteById(String id, Identity userIdentity, String source) throws
     Page page = null;
     page = getNoteById(id);
     if (page != null) {
-      if (!hasPermissionOnNote(page, PermissionType.VIEWPAGE, userIdentity)) {
+      Space space = spaceService.getSpaceByGroupId(page.getWikiOwner());
+      if (!canViewNotes(userIdentity.getUserId(), space)) {
         throw new IllegalAccessException("User does not have view the note.");
       }
-    }
-    boolean canEdit = hasPermissionOnNote(page, PermissionType.EDITPAGE, userIdentity);
-    page.setCanEdit(canEdit);
-    if(StringUtils.isNotEmpty(source)) {
-      if (source.equals("tree")) {
-        postOpenByTree(page.getWikiType(), page.getWikiOwner(), page.getName(), page);
-      }
-      if (source.equals("breadCrumb")) {
-        postOpenByBreadCrumb(page.getWikiType(), page.getWikiOwner(), page.getName(), page);
+      boolean canEdit = canCreateNotes(userIdentity.getUserId(), space);
```

Comment:
useless variable creation for canEdit and canView

Reply:
done!